### DISCUSSION TITLE
Make Tanstack Router compatible with ECMAScript Modules standard.

### DISCUSSION THREAD

**Discussion by 10thfloor** [TanStack/router#2528] [Ideas]
The release of Deno 2.0 makes me excited for the future of TypeScript development, and Tanstack will get left behind without this change. What change you ask?

The generated route file at the heart of Tanstack Router (`routeTree.gen.ts`) includes this line:

```typescript
import { Route as rootRoute } from './routes/__root'
```

With native ECMAScript modules, local import specifiers **must always include the full file extension**. It cannot be omitted.

This means Tanstack router can't be used in environments that have strict support the ECMAScript Modules standard (Deno).

Deno's approach is the correct one. Suggestion: removing the need for special workarounds (ie `moduleResolution: bundler`), and "magic" syntax, especially in auto-generated files, in favour of generating standards-compliant code!

**Comment by schiller-manuel** [comment]
are you looking for `addExtensions`?

https://tanstack.com/router/latest/docs/framework/react/guide/file-based-routing#options

![Screenshot 2024-10-13 at 08 04 32](https://github.com/user-attachments/assets/6b7872f9-8613-481a-9c10-bc00d4682d46)

**Comment by 10thfloor** [comment]
Oh, great! Why would this be an option though? Seems simpler just to include the extensions.

**Reply by schiller-manuel** [reply]
I think there were some other issues, don't remember it...
so can we close this?

**Reply by 10thfloor** [reply]
Yup :) Thanks.
